src/app/_components/product.tsx

```diff
@@ -0,0 +1,221 @@
+"use client";
+import { Switch } from "../_components/ui/switch";
+import React, { useState } from "react";
+import { Button, buttonVariants } from "../_components/ui/button";
+import Image from "next/image";
+
+interface Product {
+  id: number;
+  name: string;
+  description: string | null;
+  calification: string | null;
+  personalize: string[];
+  price: number;
+  restaurant: string | null;
+  imageUrl?: string | null;
+}
+
+interface ProductProps {
+  product: Product;
+  variant?: "default" | "mobile";
+}
+
+const Product = ({ product, variant = "default" }: ProductProps) => {
+  const { name, description, calification, personalize, price, restaurant } =
+    product || {};
+
+  const [quantity, setQuantity] = useState(0);
+
+  const decreaseQuantity = () => {
+    setQuantity((prevQuantity) => (prevQuantity > 0 ? prevQuantity - 1 : 0));
+  };
+
+  const increaseQuantity = () => {
+    setQuantity((prevQuantity) => prevQuantity + 1);
+  };
+
+  const descriptionItems =
+    description?.split("\n").filter((item) => item.trim() !== "") ?? [];
+  const [firstItem, ...restItems] = descriptionItems;
+
+  const Default = () => (
+    <div className="flex flex-col gap-8 overflow-hidden rounded-lg md:flex-row md:items-start">
+      <div className="md:w-1/2">
+        <div className="h-96 w-full rounded-b-lg rounded-t-lg bg-[url('/img/flipa-box.svg')] bg-cover"></div>
+        <div className="w-full space-y-4 p-6">
+          <h2 className="text-2xl">Descripción</h2>
+          <div className="text-lg text-gray-700">
+            {firstItem}
+            <ul className="list-inside list-disc">
+              {restItems.map((item, index) => (
+                <li key={index}>{item}</li>
+              ))}
+            </ul>
```

Comment:
<img width="693" alt="image" src="https://github.com/BrunoQuagliata1/nextjs-onboarding/assets/62408261/62120aac-55ae-49f4-9515-dd17f11a008c">

Something to keep in mind from the old react docs: https://legacy.reactjs.org/docs/lists-and-keys.html